test/blackbox-tests/test-cases/misc/run.t

```diff
@@ -1,8 +1,4 @@
   $ dune runtest --display short
-  File "dune", line 44, characters 19-42:
```

Comment:
Do you know why the test changed?

Reply:
Actually now I'm suspicious: what happens if we run an action that depends on a directory that doesn't exist?

When the directory gets created, will that action re-run or not (I assume no because I can't think of what would make it re-run or what would prevent it from running in the first case).

Reply:
There's no longer a good place to add the same check as the glob is evaluated later. I think that a more general version of this check can be added. One that will work for all predicates.

Reply:
I added a test dep-on-dir-that-does-not-exist that demonstrates the current bugs. Not sure if we should do anything about this as a part of this PR or later.